src/Agent.Worker/ExecutionContext.cs

```diff
@@ -496,10 +496,16 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
                 }
             }
 
+            message.Variables.TryGetValue("AZP_EXPAND_VULNERABLE_VARIABLES", out var expandVulnerableVariablesKnob);
+
+            string expandVulnerableVariablesKnobValue = expandVulnerableVariablesKnob?.Value ?? Environment.GetEnvironmentVariable("AZP_EXPAND_VULNERABLE_VARIABLES");
```

Comment:
>GetEnvironmentVariable [](http://example.com/codeflow?start=108&length=22)

[not blocking]

the knob already has an EnvironmentKnobSource knob source.  if that's not working it's better to figure out the root cause and address it, rather than working around

Reply:
I updated the logic, please check